src/assignments/coded-messages.rst

```diff
@@ -0,0 +1,336 @@
+      index values in the range 0 - 25.
+
+The ``build_code_dict`` Function
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Define the function ``build_code_dict`` that takes one parameter (an
+   integer).
+#. The function will build and return a dictionary. Each of the keys will be a
+   letter from the alphabet. Each of the values will also be a letter, but
+   shifted to the right a certain number of places. How many places? That
+   depends on the integer sent to the function.
+
+   *Hints*: The accumulator pattern strikes again! Also, notice how you can
+   call the ``shift_character`` function to keep your code DRY.
+
+#. The returned dictionary should include keys for BOTH lower and uppercase
+   letters.
+#. *Test your function!* In ``main()``, call ``build_code_dict`` and send in
+   different integer values. Be sure to print the returned dictionary to check
+   the results.
+
+   .. list-table:: Sample Results
+      :header-rows: 1
+
+      * - Function Call
+        - Returned Dictionary
+      * - ``build_code_dict(1)``
+        - {'a': 'b', 'b': 'c', 'c': 'd', 'd': 'e', 'e': 'f', 'f': 'g', 'g': 'h', 'h': 'i', 'i': 'j', 'j': 'k', 'k': 'l', 'l': 'm', 'm': 'n', 'n': 'o', 'o': 'p', 'p': 'q', 'q': 'r', 'r': 's', 's': 't', 't': 'u', 'u': 'v', 'v': 'w', 'w': 'x', 'x': 'y', 'y': 'z', 'z': 'a', 'A': 'B', 'B': 'C', 'C': 'D', 'D': 'E', 'E': 'F', 'F': 'G', 'G': 'H', 'H': 'I', 'I': 'J', 'J': 'K', 'K': 'L', 'L': 'M', 'M': 'N', 'N': 'O', 'O': 'P', 'P': 'Q', 'Q': 'R', 'R': 'S', 'S': 'T', 'T': 'U', 'U': 'V', 'V': 'W', 'W': 'X', 'X': 'Y', 'Y': 'Z', 'Z': 'A'}
+      * - ``build_code_dict(10)``
+        - {'a': 'k', 'b': 'l', 'c': 'm', 'd': 'n', 'e': 'o', 'f': 'p', 'g': 'q', 'h': 'r', 'i': 's', 'j': 't', 'k': 'u', 'l': 'v', 'm': 'w', 'n': 'x', 'o': 'y', 'p': 'z', 'q': 'a', 'r': 'b', 's': 'c', 't': 'd', 'u': 'e', 'v': 'f', 'w': 'g', 'x': 'h', 'y': 'i', 'z': 'j', 'A': 'K', 'B': 'L', 'C': 'M', 'D': 'N', 'E': 'O', 'F': 'P', 'G': 'Q', 'H': 'R', 'I': 'S', 'J': 'T', 'K': 'U', 'L': 'V', 'M': 'W', 'N': 'X', 'O': 'Y', 'P': 'Z', 'Q': 'A', 'R': 'B', 'S': 'C', 'T': 'D', 'U': 'E', 'V': 'F', 'W': 'G', 'X': 'H', 'Y': 'I', 'Z': 'J'}
+      * - ``build_code_dict(33)``
+        - {'a': 'h', 'b': 'i', 'c': 'j', 'd': 'k', 'e': 'l', 'f': 'm', 'g': 'n', 'h': 'o', 'i': 'p', 'j': 'q', 'k': 'r', 'l': 's', 'm': 't', 'n': 'u', 'o': 'v', 'p': 'w', 'q': 'x', 'r': 'y', 's': 'z', 't': 'a', 'u': 'b', 'v': 'c', 'w': 'd', 'x': 'e', 'y': 'f', 'z': 'g', 'A': 'H', 'B': 'I', 'C': 'J', 'D': 'K', 'E': 'L', 'F': 'M', 'G': 'N', 'H': 'O', 'I': 'P', 'J': 'Q', 'K': 'R', 'L': 'S', 'M': 'T', 'N': 'U', 'O': 'V', 'P': 'W', 'Q': 'X', 'R': 'Y', 'S': 'Z', 'T': 'A', 'U': 'B', 'V': 'C', 'W': 'D', 'X': 'E', 'Y': 'F', 'Z': 'G'}
+
+.. admonition:: Note
+
+   Remember that dictionaries are *unordered* collections. The key/value pairs
+   in your results might be in a different order than the examples, and that is
+   OK.
+
+Part 2: Create Coded Messages
```

Comment:
Can this title change to Create Coded and Translated Messages? or Code and Decode Messages? something to include the decrypt function ask as well